src/move/econia/sources/market.move

```diff
@@ -628,6 +613,34 @@ module econia::market {
         map: Tablist<u64, OrderBook>
     }
 
+    struct MarketEventHandles has key {
+        /// Not emitted at user level when swapper doesn't sign.
+        place_swap_order_events:
+            Tablist<u64, EventHandle<PlaceSwapOrderEvent>>
```

Comment:
The market IDs are not necessarily sequential since a market that was initialized before an on-chain upgrade will initialize an event handle on-the-fly.

However the major drawback I see to switching away from a table-based approach is that storage creations/writes are fixed-cost up to a critical number of bytes (2k at the moment I believe). Aptos Labs discourages vectors for "large" datasets, because the cost to operate on the vector increases linearly for each additional entry.

I believe each `EventHandle` is 48 bytes, so after 42 markets or so get initialized the cost to emit an event grows linearly for every additional market registered. This situation introduces a DoS attack, which is emitted by a Table-based approach.

The rationale for using a `Tablist` instead of a `Table` is to enable iteration over all handles using the Aptos REST API.

Your thoughts?

Reply:
> The market IDs are not necessarily sequential since a market that was initialized before an on-chain upgrade will initialize an event handle on-the-fly.

Explain? The market IDs are set here, right. How would this ever produce non-sequential IDs?

https://github.com/econia-labs/econia/blob/b227eb35bbd8c4c350336f29bdd72cc02d65b8e5/src/move/econia/sources/registry.move#L1593-L1594

Reply:
> > The market IDs are not necessarily sequential since a market that was initialized before an on-chain upgrade will initialize an event handle on-the-fly.
> 
> Explain? The market IDs are set here, right. How would this ever produce non-sequential IDs?
> 
> https://github.com/econia-labs/econia/blob/b227eb35bbd8c4c350336f29bdd72cc02d65b8e5/src/move/econia/sources/registry.move#L1593-L1594

@bogberry per https://github.com/econia-labs/econia/pull/321#discussion_r1261206551

> The market IDs are not necessarily sequential since a market that was initialized before an on-chain upgrade will initialize an event handle on-the-fly.

The market IDs are sequential but the order of initialization of event handles isn't necessarily sequential. E.g. markets 1-5 as legacy markets, the event handle init sequence could look like: 6,7,8,1,2,3,4,5,9,10,...

Reply:
Oh, by "on-chain upgrade" you mean the changes in this PR.

In what situation would this matter though? If markets 1-5 were created before this change, and markets 6-8 were initialized prior to event handles being created for 1-5, when would you want to traverse these event handles in the order 6,7,8,1,2,3,4,5,9,10,...?

Reply:
@bogberry per https://github.com/econia-labs/econia/pull/321#issuecomment-1635034144 `Tablist` updated to `Table`